src/core/stream/representation/get_needed_segments.ts

```diff
@@ -103,8 +114,32 @@ export default function getNeededSegments({
   getBufferedHistory,
   neededRange,
   segmentsBeingPushed,
+  maxVideoBufferSize,
 } : IGetNeededSegmentsArguments) : ISegment[] {
   const { representation } = content;
+  const estimateSizeBeingPushed = segmentsBeingPushed.reduce((size, segment) => {
+    const { bitrate } = segment.representation;
+    // Not taking into account the fact that the segment
+    // can still be generated and the duration not fully exact
+    const { duration } = segment.segment;
+    return size + (bitrate * duration);
+  }, 0);
+  let availableBufferSize = bufferedSegments.reduce((size, chunk) => {
+    if (chunk.chunkSize !== undefined) {
+      return size - chunk.chunkSize;
+    } else {
+      return size;
+    }
+
+  } , maxVideoBufferSize) - estimateSizeBeingPushed;
```

Comment:
As we're in a subject of factoring code, both could be put in another function no?
Something like `getCurrentlyUsedBufferSize` / `getRemainingBufferSize` depending on what it returns